amazon_cloud_code_generator/data/plugins/module_utils/core.py

```diff
@@ -467,10 +477,10 @@ def update_resource(
                 ) as e:
                     self.module.fail_json_aws(e, msg="Failed to update resource")
 
-                if self.module.params.get("wait"):
-                    self.wait_until_resource_request_success(
-                        response["ProgressEvent"]["RequestToken"]
-                    )
+                # Always wait to update to ensure correct data returned to user
```

Comment:
We cannot always wait for updates. It there's an issue with tag update, we should handle singularly!